actions/docs-verifier/src/RedirectionVerifier/WhatsNewConfigurationReader.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using System.Text.Json;
+
+namespace RedirectionVerifier
+{
+    public static class WhatsNewConfigurationReader
+    {
+        private static readonly JsonSerializerOptions s_options = new() { AllowTrailingCommas = true };
+        private static WhatsNewConfiguration s_cachedWhatsNewConfiguration = null!;
+        private const string WhatsNewConfigurationFileName = ".whatsnew.json";
+
+        /// <summary>
+        /// Retrieves the configured "What's new" directory from <c>.whatsnew.json</c>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Failed to read <c>.whatsnew.json</c>.</exception>
+        public static string? GetWhatsNewPath()
```

Comment:
One concern about this, as written.

If a repo doesn't have a `.whatsnew.json`, the code will never cache its non-existence. It will only cost the `File.Exists` call, but depending on the PR, that might matter...

Nit, but something to think about.